tests/beta/interfaces/test_tape_tf.py

```diff
@@ -785,3 +783,571 @@ def test_sampling(self):
+)
+class TestQNode:
+    """Same tests as above, but this time via the QNode interface!"""
+
+    def test_execution_no_interface(self, dev_name, diff_method):
+        """Test execution works without an interface"""
+        if diff_method == "backprop":
+            pytest.skip("Test does not support backprop")
+
+        dev = qml.device(dev_name, wires=1)
+
+        @qnode(dev)
+        def circuit(a):
+            qml.RY(a, wires=0)
+            qml.RX(0.2, wires=0)
+            return expval(qml.PauliZ(0))
+
+        a = tf.Variable(0.1)
+
+        with tf.GradientTape() as tape:
+            res = circuit(a)
+
+        assert circuit.qtape.interface == "autograd"
+
+        # without the interface, the tape simply returns an array of results
+        assert isinstance(res, np.ndarray)
+        assert res.shape == (1,)
+        assert isinstance(res[0], float)
+
+        # without the interface, the tape is unable to deduce
+        # trainable parameters
+        assert circuit.qtape.trainable_params == {0, 1}
+
+        # gradients should cause an error
+        with pytest.raises(AttributeError, match="has no attribute '_id'"):
+            assert tape.gradient(res, a) is None
+
+    def test_execution_with_interface(self, dev_name, diff_method):
+        """Test execution works with the interface"""
+        if diff_method == "backprop":
+            pytest.skip("Test does not support backprop")
+
+        dev = qml.device(dev_name, wires=1)
+
+        @qnode(dev, interface="tf")
+        def circuit(a):
+            qml.RY(a, wires=0)
+            qml.RX(0.2, wires=0)
+            return expval(qml.PauliZ(0))
+
+        a = tf.Variable(0.1)
+        circuit(a)
+
+        # if executing outside a gradient tape, the number of trainable parameters
+        # cannot be determined by TensorFlow
+        assert circuit.qtape.trainable_params == set()
+
+        with tf.GradientTape() as tape:
+            res = circuit(a)
+
+        assert circuit.qtape.interface == "tf"
+
+        # with the interface, the tape returns tensorflow tensors
+        assert isinstance(res, tf.Tensor)
+        assert res.shape == (1,)
+
+        # the tape is able to deduce trainable parameters
+        assert circuit.qtape.trainable_params == {0}
+
+        # gradients should work
+        grad = tape.gradient(res, a)
+        assert isinstance(grad, tf.Tensor)
+        assert grad.shape == tuple()
+
+    def test_interface_swap(self, dev_name, diff_method, tol):
+        """Test that the TF interface can be applied to a QNode
+        with a pre-existing interface"""
+        if diff_method == "backprop":
+            pytest.skip("Test does not support backprop")
+
+        dev = qml.device(dev_name, wires=1)
+
+        @qnode(dev, interface="autograd")
+        def circuit(a):
+            qml.RY(a, wires=0)
+            qml.RX(0.2, wires=0)
+            return expval(qml.PauliZ(0))
+
+        from pennylane import numpy as anp

```

Comment:
How come this import is placed here?

Reply:
Ah, autograd is only import for this single test! Since the tests aren't linted, I thought it was safer to keep this import close to the test